### DISCUSSION TITLE
Programmatically Create Prisma schema,

### DISCUSSION THREAD

**Discussion by cpt-n3mo** [prisma/prisma#12479] [Q&A] [answered]
Good day!

I'm new to Prisma, but for what i have seen it will make my DX allot nicer that's for sure !
I have been wanting to build a wrapper around prisma, and would like to dynamically generate a schema based on a config
is there a possibility to generate the Prisma schema from a api..?
or would generating it from strings be the the solution for this..?

kind regards!

**Comment by austincrim** [comment, marked as answer]
Hey @cpt-n3mo 👋, we are glad you are enjoying Prisma!

We don't have any official solutions for generating a Prisma Schema. You are correct that you would have to build it up yourself in code.

What is the purpose of the wrapper you are creating?

**Reply by cpt-n3mo** [reply]
good day @austincrim

yeh its pritty nice to work with very intuitive, even though i kind of have a love / hate relationship with typescript :rofl: 
and that is kind of a pain though. for the reason of the wrapper:

i want to create a module for nuxt3 as this comes with a nitro server now a days.!
i would like to integrate prisma on the server side with this wrapper that feeds a CRUD route generator.
it should also pass data so that the admin panel knows what to render so i can add, edit and remove data,   
but for this is would be nice if i could generate the schemas dynamically based on my own schema model..

i will have to take a look, generating the models with strings is maybe going to be abit error prone, i did see there was a way to generate a json with a generator, so i might take a look on how this is setup so i can maybe reverse the process in a compatible way.. 
at least thanks for your information this means i can stop searching the web en start programming :rofl: 

regards!

**Comment by sixman9** [comment]
JFYI:

https://github.com/cwqt/refract => Generate Prisma from TypeScript

(I maintain ***no*** affiliation with the developer(s)).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
